test/logoutresponse_test.rb

```diff
@@ -70,42 +155,160 @@ class RubySamlTest < Minitest::Test
         logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response({:uuid => in_relation_to_request_id}), settings)
 
         logoutresponse.validate!
+        assert_empty logoutresponse.errors
       end
 
-      it "raises validation error when matching for wrong request id" do
+      it "raises validation error when response initiated with blank" do
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new("", settings)
+
+        expected_error_msg = "Blank Logout Response"
+        assert_raises(OneLogin::RubySaml::ValidationError, expected_error_msg) do
+          logoutresponse.validate!
+        end
+        assert logoutresponse.errors.include? expected_error_msg
+      end
 
+      it "raises validation error when matching for wrong request id" do
         expected_request_id = "_some_other_expected_id"
-        opts = { :matches_request_id => expected_request_id}
 
-        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response, settings, opts)
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response, settings)
 
-        assert_raises(OneLogin::RubySaml::ValidationError) { logoutresponse.validate! }
+        expected_error_msg = "Logout Response does not match the request ID, expected: <#{expected_request_id}>, but was: <#{logoutresponse.in_response_to}>"
+        assert_raises(OneLogin::RubySaml::ValidationError, expected_error_msg) do
+          logoutresponse.validate!(false, expected_request_id)
+        end
+        assert logoutresponse.errors.include? expected_error_msg
       end
 
       it "raise validation error for wrong request status" do
         logoutresponse = OneLogin::RubySaml::Logoutresponse.new(unsuccessful_response, settings)
 
-        assert_raises(OneLogin::RubySaml::ValidationError) { logoutresponse.validate! }
+        expected_error_msg = "The status code of the Logout Response was not Success, was Requester"
+        assert_raises(OneLogin::RubySaml::ValidationError, expected_error_msg) do
+          logoutresponse.validate!
+        end
+        assert logoutresponse.errors.include? expected_error_msg
+      end
+
+      it "raise validation error for wrong request status and status_message" do
+        unsuccessful_response_with_status_message = unsuccessful_response
+        unsuccessful_response_with_status_message = unsuccessful_response_with_status_message.gsub('</samlp:Status>', '<samlp:StatusMessage>It was requester</samlp:StatusMessage></samlp:Status>')
```

Comment:
Instead of creating a new unsuccessful_response example file, I took an existed example and changed it, making it a unsuccessful_response_with_status_message.